packages/clarity-js/src/data/metadata.ts

```diff
@@ -144,9 +144,13 @@ function track(u: User, consent: BooleanFlag = null): void {
   // Convert time precision into days to reduce number of bytes we have to write in a cookie
   // E.g. Math.ceil(1628735962643 / (24*60*60*1000)) => 18852 (days) => ejo in base36 (13 bytes => 3 bytes)
   let end = Math.ceil((Date.now() + (Setting.Expire * Time.Day))/Time.Day);
+  // If dob is not set in the user object, use today's date as a dob
+  let date = new Date();
+  let dob = [date.getFullYear(), ('0' + (date.getMonth() + 1)).slice(-2), ('0' + date.getDate()).slice(-2)].join('');
+  dob = u.dob === null ? dob : u.dob;
   // To avoid cookie churn, write user id cookie only once every day
-  if (u.expiry === null || Math.abs(end - u.expiry) >= Setting.CookieInterval || u.consent !== consent) {
-    setCookie(Constant.CookieKey, [data.userId, Setting.CookieVersion, end.toString(36), consent].join(Constant.Pipe), Setting.Expire);
+  if (u.expiry === null || Math.abs(end - u.expiry) >= Setting.CookieInterval || u.consent !== consent || u.dob !== dob) {
+    setCookie(Constant.CookieKey, [data.userId, Setting.CookieVersion, end.toString(36), consent, dob].join(Constant.Pipe), Setting.Expire);
```

Comment:
Check that new cookie will work with older code. Can't assume that older client doesn't get new cookie.

Reply:
Yeah I tested that, the older client code will read the first 4 parts of the cookie and ignore the dob. Similarly with the server code, will just ignore the DOB field of the envelope.